### PR TITLE
chore(hardware):  integration tests for serial number interaction

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Integration tests for serial number setting and reading.

# Changelog

# Review requests

None

# Risk assessment

None

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

File: hardware/opentrons_hardware/firmware_bindings/messages/payloads.py [hunk 1]
```diff
@@ -375,9 +375,9 @@ class BindSensorOutputResponsePayload(utils.BinarySerializable):
 class PipetteInfoResponsePayload(utils.BinarySerializable):
```
Comment (Laura-Danielle) [inline hardware/opentrons_hardware/firmware_bindings/messages/payloads.py:376]: This is going to be deleted in alise's PR so I would refrain from making this change, unless it's necessary.

**Review by Laura-Danielle — CHANGES_REQUESTED** [review]
Maybe you and @ahiuchingau  can coordinate some how? This is pretty directly related to [this pr](https://github.com/Opentrons/opentrons/pull/10557).

File: hardware/opentrons_hardware/firmware_bindings/messages/payloads.py (hunk 1 above)
Comment (ahiuchingau) [reply hardware/opentrons_hardware/firmware_bindings/messages/payloads.py:376]: I'm not deleting the responses, just replacing `PipetteInfoRequest` and `GripperInfoRequest` with `InstrumentInfoRequest`. The responses are unchanged

Comment (amitlissack) [reply hardware/opentrons_hardware/firmware_bindings/messages/payloads.py:376]: Tell me what to do and I'll do it!!!

Comment (amitlissack) [reply hardware/opentrons_hardware/firmware_bindings/messages/payloads.py:376]: Or... if there's a FW dependency that isn't quite ready, @ahiuchingau  maybe you can take over this PR?

Comment (ahiuchingau) [reply hardware/opentrons_hardware/firmware_bindings/messages/payloads.py:376]: Oh what I meant to say was there shouldn't be any conflict. I'm happy to take over, maybe rebasing would be less of a pain that way!

**Review by ahiuchingau — APPROVED** [review]
LGTM!

**Review by Laura-Danielle — APPROVED** [review]

**Pull request merged**